cmd/jaeger/internal/integration/kafka_test.go

```diff
@@ -20,30 +60,61 @@ func TestKafkaStorage(t *testing.T) {
 	// we can change the config to use topic: "${KAFKA_TOPIC:-jaeger-spans}"
 	// and export a KAFKA_TOPIC var with random topic name in the tests.
 
-	collectorConfig := "../../collector-with-kafka.yaml"
-	ingesterConfig := "../../ingester-remote-storage.yaml"
+	// OTLP config files
+	baseCollectorConfig := "../../collector-with-kafka.yaml"
+	baseIngesterConfig := "../../ingester-remote-storage.yaml"
 
-	collector := &E2EStorageIntegration{
-		SkipStorageCleaner:  true,
-		ConfigFile:          collectorConfig,
-		HealthCheckEndpoint: "http://localhost:8888/metrics",
-	}
+	// Test OTLP formats (no need to modify as default is otlp_proto)
+	t.Run("OTLPFormats", func(t *testing.T) {
+		t.Log("Starting OTLPFormats test")
+		collector := &E2EStorageIntegration{
+			SkipStorageCleaner:  true,
+			ConfigFile:          baseCollectorConfig,
+			HealthCheckEndpoint: "http://localhost:8888/metrics",
+		}
+		collector.e2eInitialize(t, "kafka")
+		t.Log("Collector initialized")
 
-	// Initialize and start the collector
-	collector.e2eInitialize(t, "kafka")
+		ingester := &E2EStorageIntegration{
+			ConfigFile: baseIngesterConfig,
+			StorageIntegration: integration.StorageIntegration{
+				CleanUp:                      purge,
+				GetDependenciesReturnsSource: true,
+				SkipArchiveTest:              true,
+			},
+		}
+		ingester.e2eInitialize(t, "kafka")
+		t.Log("Ingester initialized")
+		ingester.RunSpanStoreTests(t)
+		t.Log("OTLPFormats test completed successfully")
+	})
 
-	ingester := &E2EStorageIntegration{
-		ConfigFile: ingesterConfig,
-		StorageIntegration: integration.StorageIntegration{
-			CleanUp:                      purge,
-			GetDependenciesReturnsSource: true,
-			SkipArchiveTest:              true,
-		},
-	}
+	// Test legacy Jaeger formats
+	t.Run("LegacyJaegerFormats", func(t *testing.T) {
```

Comment:
I suggest doing a table-driven test where you loop through multiple formats (otlp_proto, otlp_json, jaeger_***). This way you don't repeat the actual code, always go through replacement logic (even when replacing otlp_proto -> otlp_proto).

Reply:
Okay understood.